src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs

```diff
@@ -87,7 +87,7 @@ public static ExpressionsAnalysis ProcessSemanticModel(MongoAnalyzerContext cont
                     typesProcessor.ProcessTypeSymbol(typeArgument);
                 }
 
-                var (newBuildersExpression, constantsMapper) = RewriteBuildersExpression(node, typesProcessor, semanticModel);
+                var (newBuildersExpression, constantsMapper) = RewriteBuildersExpression(builderExpressionNode, typesProcessor, semanticModel);
```

Comment:
Looks like this change is sufficient, without changes in `IsValidBuildersExpression `.

Reply:
Done